Ah! Interesting...

```
=== FAIL: libnetwork/drivers/bridge TestLinkContainers (0.06s)
time="2023-07-16T18:55:00Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge/net1/ is not added to the store"
time="2023-07-16T18:55:00Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge-endpoint/ep1/ is not added to the store"
time="2023-07-16T18:55:00Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge-endpoint/ep1/ is not added to the store"
time="2023-07-16T18:55:00Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge-endpoint/ep2/ is not added to the store"
time="2023-07-16T18:55:00Z" level=warning msg="bridge store not initialized. kv object docker/network/v1.0/bridge-endpoint/ep2/ is not added to the store"
panic: iptables.IPVersion("unknown IP version: ") [recovered]
	panic: iptables.IPVersion("unknown IP version: ")

goroutine 515 [running]:
testing.tRunner.func1.2({0x25eb40, 0x40008709c0})
	/usr/local/go/src/testing/testing.go:1526 +0x1c8
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x364
panic({0x25eb40, 0x40008709c0})
	/usr/local/go/src/runtime/panic.go:884 +0x1f4
github.com/docker/docker/libnetwork/iptables.GetIptable(...)
	/go/src/github.com/docker/docker/libnetwork/iptables/iptables.go:166
github.com/docker/docker/libnetwork/iptables.(*ChainInfo).Link(0x40006a0890?, {0x2b6b75?, 0x400021f458?}, {0x40006a08a0?, 0x400021f3d8?, 0x1cdf4?}, {0x40006a08b0?, 0xffffa07815b8?, 0x10?}, 0x493380?, ...)
	/go/src/github.com/docker/docker/libnetwork/iptables/iptables.go:386 +0x2c4
github.com/docker/docker/libnetwork/drivers/bridge.linkContainers({0x2b6b75?, 0x9054?}, {0x40006a0880, 0xb}, {0x40006a0890, 0xb}, {0x40006a04c0, 0x3, 0x400021f4f8?}, {0x2b7b67, ...}, ...)
	/go/src/github.com/docker/docker/libnetwork/drivers/bridge/link.go:80 +0x4d4
github.com/docker/docker/libnetwork/drivers/bridge.(*link).Enable.func1(...)
	/go/src/github.com/docker/docker/libnetwork/drivers/bridge/link.go:38
github.com/docker/docker/libnetwork/drivers/bridge.(*link).Enable(0x40008dc820)
	/go/src/github.com/docker/docker/libnetwork/drivers/bridge/link.go:42 +0x2e0
github.com/docker/docker/libnetwork/drivers/bridge.(*driver).link(0x400021f940?, 0x40009c8b40, 0x40009943c0, 0x1)
	/go/src/github.com/docker/docker/libnetwork/drivers/bridge/bridge.go:1418 +0x63c
github.com/docker/docker/libnetwork/drivers/bridge.(*driver).ProgramExternalConnectivity(0x40008a2510, {0x2b7002?, 0x2bebb7?}, {0x2b6d70, 0x3}, 0x316180?)
	/go/src/github.com/docker/docker/libnetwork/drivers/bridge/bridge.go:1317 +0x6ac
github.com/docker/docker/libnetwork/drivers/bridge.TestLinkContainers(0x40008b4000)
	/go/src/github.com/docker/docker/libnetwork/drivers/bridge/bridge_test.go:785 +0xbb0
testing.tRunner(0x40008b4000, 0x2ce1e8)
	/usr/local/go/src/testing/testing.go:1576 +0x104
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1629 +0x370

DONE 311 tests, 2 skipped, 1 failure in 185.496s

```

--- a/libnetwork/iptables/iptables.go
+++ b/libnetwork/iptables/iptables.go
@@ -144,8 +144,18 @@ func initCheck() error {
 	return nil
 }
 
-// GetIptable returns an instance of IPTable with specified version
+// GetIptable returns an instance of IPTable with specified version ([IPv4]
+// or [IPv6]). It panics if an invalid [IPVersion] is provided.
 func GetIptable(version IPVersion) *IPTable {
+	switch version {
+	case IPv4, IPv6:
+		// valid version
+	case "":
+		// default is IPv4 for backward-compatibility
+		version = IPv4
+	default:
+		panic("unknown IP version: " + version)
+	}
 	return &IPTable{ipVersion: version}
 }
 

--- a/libnetwork/iptables/iptables.go
+++ b/libnetwork/iptables/iptables.go
@@ -94,6 +94,19 @@ func (e ChainError) Error() string {
 	return fmt.Sprintf("error iptables %s: %s", e.Chain, string(e.Output))
 }
 
+// loopbackAddress returns the loopback address for the given IP version.
+func loopbackAddress(version IPVersion) string {
+	switch version {
+	case IPv4, "":
+		// IPv4 (default for backward-compatibility)
+		return "127.0.0.0/8"
+	case IPv6:
+		return "::1/128"
+	default:
+		panic("unknown IP version: " + version)
+	}
+}
+
 func detectIptables() {
 	path, err := exec.LookPath("iptables")
 	if err != nil {
@@ -183,14 +196,6 @@ func (iptable IPTable) NewChain(name string, table Table, hairpinMode bool) (*Ch
 	}, nil
 }
 
-// LoopbackByVersion returns loopback address by version
-func (iptable IPTable) LoopbackByVersion() string {
-	if iptable.ipVersion == IPv6 {
-		return "::1/128"
-	}
-	return "127.0.0.0/8"
-}
-
 // ProgramChain is used to add rules to a chain
 func (iptable IPTable) ProgramChain(c *ChainInfo, bridgeName string, hairpinMode, enable bool) error {
 	if c.Name == "" {
@@ -232,7 +237,7 @@ func (iptable IPTable) ProgramChain(c *ChainInfo, bridgeName string, hairpinMode
 			"-j", c.Name,
 		}
 		if !hairpinMode {
-			output = append(output, "!", "--dst", iptable.LoopbackByVersion())
+			output = append(output, "!", "--dst", loopbackAddress(iptable.ipVersion))
 		}
 		if !iptable.Exists(Nat, "OUTPUT", output...) && enable {
 			if err := c.Output(Append, output...); err != nil {
@@ -443,15 +448,15 @@ func (c *ChainInfo) Output(action Action, args ...string) error {
 
 // Remove removes the chain.
 func (c *ChainInfo) Remove() error {
-	iptable := GetIptable(c.IPVersion)
 	// Ignore errors - This could mean the chains were never set up
 	if c.Table == Nat {
 		_ = c.Prerouting(Delete, "-m", "addrtype", "--dst-type", "LOCAL", "-j", c.Name)
-		_ = c.Output(Delete, "-m", "addrtype", "--dst-type", "LOCAL", "!", "--dst", iptable.LoopbackByVersion(), "-j", c.Name)
+		_ = c.Output(Delete, "-m", "addrtype", "--dst-type", "LOCAL", "!", "--dst", loopbackAddress(c.IPVersion), "-j", c.Name)
 		_ = c.Output(Delete, "-m", "addrtype", "--dst-type", "LOCAL", "-j", c.Name) // Created in versions <= 0.1.6
 		_ = c.Prerouting(Delete)
 		_ = c.Output(Delete)
 	}
+	iptable := GetIptable(c.IPVersion)
 	_, _ = iptable.Raw("-t", string(c.Table), "-F", c.Name)
 	_, _ = iptable.Raw("-t", string(c.Table), "-X", c.Name)
 	return nil

